core/src/main/kotlin/org/evomaster/core/problem/rest/service/AIResponseClassifier.kt

```diff
@@ -11,46 +11,38 @@ import org.evomaster.core.problem.rest.classifier.GaussianOnlineClassifier
 import javax.annotation.PostConstruct
 
 
+
 class AIResponseClassifier : AIModel {
 
     @Inject
-    private lateinit var config : EMConfig
+    private lateinit var config: EMConfig
 
-    private lateinit var delegate: AIModel
+    private lateinit var model: AIModel
 
     @PostConstruct
-    fun initModel(){
-
-        val dimension = 1 //FIXME
-
-        when(config.aiModelForResponseClassification){
-            EMConfig.AIResponseClassifierModel.GAUSSIAN -> {
-                delegate = GaussianOnlineClassifier(dimension)
-            }
-            EMConfig.AIResponseClassifierModel.GLM -> {
-                delegate = GLMOnlineClassifier(dimension, config.aiResponseClassifierLearningRate)
-            }
-            EMConfig.AIResponseClassifierModel.NN -> {
-                //TODO
-            }
-            EMConfig.AIResponseClassifierModel.NONE -> {
-                //TODO
-                delegate = object : AIModel {
-                    override fun updateModel(input: RestCallAction, output: RestCallResult) {}
-                    override fun classify(input: RestCallAction) = AIResponseClassification()
-                }
+    fun initModel() {
+        model = when (config.aiModelForResponseClassification) {
+            EMConfig.AIResponseClassifierModel.GAUSSIAN ->
+                GaussianOnlineClassifier()
+            EMConfig.AIResponseClassifierModel.GLM ->
+                GLMOnlineClassifier(config.aiResponseClassifierLearningRate)
+            else -> object : AIModel {
+                override fun updateModel(input: RestCallAction, output: RestCallResult) {}
+                override fun classify(input: RestCallAction) = AIResponseClassification()
             }
         }
     }
 
+
     override fun updateModel(input: RestCallAction, output: RestCallResult) {
-        delegate.updateModel(input, output)
+        model.updateModel(input, output)
     }
 
     override fun classify(input: RestCallAction): AIResponseClassification {
-        return delegate.classify(input)
+        return model.classify(input)
     }
 
+    fun getInnerModel(): AIModel = model
```

Comment:
returning a mutable state reference is usually not a good idea in OO, as it can lead break incapsulation. at times, we need it. in those cases, to make it explicit, i prefer to use the term `view` instead of `get`, eg, `viewInnerModel`